File: app-mvvm/src/main/java/com/xuie/imaginaryandroid/gui/web/WebViewActivity.java
package com.xuie.imaginaryandroid.gui.web;

import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.View;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import com.xuie.imaginaryandroid.R;
import com.xuie.imaginaryandroid.base.BaseActivity;
import com.xuie.imaginaryandroid.databinding.ActivityWebviewBinding;

/**
 * @author xuie
 */
public class WebViewActivity extends BaseActivity {

    public static final String URL = "url";

    @Override
    protected int getLayoutId() {
        return R.layout.activity_webview;
    }

    @Override
    protected void onInit(@Nullable Bundle savedInstanceState) {
        ActivityWebviewBinding binding = getDataBinding();
        String url = getIntent().getStringExtra(URL);

        // Configure related browser settings
        binding.webView.getSettings().setLoadsImagesAutomatically(true);
        binding.webView.getSettings().setJavaScriptEnabled(true);
        binding.webView.setScrollBarStyle(View.SCROLLBARS_INSIDE_OVERLAY);
        // Configure the client to use when opening URLs
        binding.webView.setWebViewClient(new MyBrowser());
        // Load the initial URL
        binding.webView.loadUrl(url);
    }

    @Override
    protected View[] setImmersiveView() {
        return new View[0];
    }

    /**
     * Manages the behavior when URLs are loaded
     */
    private class MyBrowser extends WebViewClient {
        @Override
        public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
            view.loadUrl(request.getUrl().toString());
            return super.shouldOverrideUrlLoading(view, request);
        }
    }

    @Override
    public void onBackPressed() {
//        if (webView.canGoBack()) {
//            webView.goBack()
//        } else {
        super.onBackPressed();
//        }
    }
}

File: app-mvvm-kotlin/src/main/kotlin/xuk/imaginary/gui/web/WebViewActivity.kt
package xuk.imaginary.gui.web

import android.annotation.SuppressLint
import android.os.Bundle
import android.view.View
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import xuk.imaginary.R
import xuk.imaginary.databinding.ActivityWebviewBinding

/**
 * @author Jie Xu
 */
class WebViewActivity : AppCompatActivity() {

  @SuppressLint("SetJavaScriptEnabled")
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    val binding = DataBindingUtil.setContentView<ActivityWebviewBinding>(this, R.layout.activity_webview)

    val url = intent.getStringExtra(URL)

    // Configure related browser settings
    binding.webView.settings.loadsImagesAutomatically = true
    binding.webView.settings.javaScriptEnabled = true
    binding.webView.scrollBarStyle = View.SCROLLBARS_INSIDE_OVERLAY
    // Configure the client to use when opening URLs
    binding.webView.webViewClient = MyBrowser()
    // Load the initial URL
    binding.webView.loadUrl(url)
  }

  /**
   * Manages the behavior when URLs are loaded
   */
  private inner class MyBrowser : WebViewClient() {
    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
      view.loadUrl(request.url.toString())
      return super.shouldOverrideUrlLoading(view, request)
    }
  }

  companion object {
    const val URL = "url"
  }
}
